artiq/frontend/artiq_client.py

```diff
@@ -291,6 +300,8 @@ def main():
             _show_dict(args, "devices", _show_devices)
         elif args.what == "datasets":
             _show_dict(args, "datasets", _show_datasets)
+        else:
+            raise ValueError("Unknown object to show; list valid names with -h")
```

Comment:
Wrong exception message. The user is never supposed to see this; if this code is reached it is a bug in the program and using ``-h`` is not going to hep.

Reply:
Same error message as was there before
https://github.com/m-labs/artiq/blob/6df4ae934f07e57f6e61be1bafacf91179b40b01/artiq/frontend/artiq_client.py#L301-L303

Reply:
Sure, without ``choices``.

Reply:
This should be caught using argparse's ``choices`` kwarg, so removed entirely. If you don't want that, suggest what error message would be better.
https://github.com/m-labs/artiq/pull/1185/commits/a3711a4e0c1f942f8800f2981a9f7a9652aebfbc

Reply:
No error message is fine since users are never supposed to see this error. This is only to give a hint to the developer.

Reply:
> This should be caught using argparse's `choices` kwarg, so removed entirely.

Yes, if the program is bug-free; as we all know, they always are.

Reply:
And I already explained this https://github.com/m-labs/artiq/pull/1185#discussion_r230968686